components/brave_new_tab_ui/reducers/cryptoDotCom_reducer.ts

```diff
@@ -15,51 +15,54 @@ function performSideEffect (fn: () => void): void {
   window.setTimeout(() => fn(), 0)
 }
 
+function reducePairs (rawPairs: SupportedPair[]) {
+  return rawPairs.reduce((pairs: object, currPair: SupportedPair) => {
+    const { base, pair } = currPair
+    pairs[base] = pairs[base]
+      ? [...pairs[base], pair]
+      : [pair]
+    return pairs
+  }, {})
+}
+
 const cryptoDotComReducer: Reducer<NewTab.State | undefined> = (state: NewTab.State, action) => {
   const payload = action.payload
 
   switch (action.type) {
-    case types.ON_TOTAL_PRICE_OPT_IN:
-      state = { ...state }
-      state.cryptoDotComState.optInTotal = true
-      break
-
     case types.ON_BTC_PRICE_OPT_IN:
       state = { ...state }
       state.cryptoDotComState.optInBTCPrice = true
       break
 
-    case types.SET_TICKER_PRICES:
+    case types.SET_MARKETS_REQUESTED:
       state = { ...state }
       state.cryptoDotComState.tickerPrices = {
         ...state.cryptoDotComState.tickerPrices,
-        ...payload
+        ...payload.tickerPrices
       }
+      state.cryptoDotComState.losersGainers = payload.losersGainers
```

Comment:
This is invalid for redux since we are only changing properties and reusing the same parent object, so could result in a non-render even though data has changed. Yes, `oldState !== newState` so the root of the component tree will start to render, if a pure component is only passed `state.cryptoDotComState` as a prop, `oldCryptoDotComState === newCryptoDotComState`, so no render would happen. Instead we would do something like:

```js
state.cryptoDotComState = {
   ...state.cryptoDotComState,
   tickerPrices: {
      ...state.cryptoDotComState.tickerPrices,
      ...payload.tickerPrices,
   },
   losersGainers: payload.losersGainers,
}
```
...or whichever method looks cleaner to you

Reply:
fixed